### PR TITLE
For #8143: Allow add-on popups to display prompts

### LINKED ISSUE #8143: [Bug]uBlock settings freeze after tapping the "Reset to default settings" button
## Prerequisites
Have uBlock Origin installed.

## Steps to reproduce
1. From the Add-ons manager tap on uBlock Origins
2. Tap and open "Settings"
3. Swipe down Settings section tap the  "Reset to default settings" button

### Expected behavior
A confirmation prompt asking the user to confirm the reset of Ublock to factory settings should be displayed.
After the user confirms the reset, he should be able to access the other Settings sections (e.g. "File lists, "My filters" etc.)  

### Actual behavior
After tapping the button you can't access the other Settings sections (e.g. "File lists, "My filters" etc.)  

### Device information

* Android device: 
• Google Pixel 3a XL (Android 10)
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7.0.0)
• OnePlus A3 (Android 6.0.1)
• LG Nexus 4 (Android 5.1.1)

* Fenix version: Nightly #20370604 from 2/6

### Notes
<details>
    <summary>Video comparison between Fennec VS Fenix (dropdown)</summary>

► Fenix

►Fennec

</details>

### PR DESCRIPTION
**Pull request by csadilek**
This is a refactoring to introduce a base class for our add-on popup fragments. The base class now also brings in the `PromptFeature` to fix #8143.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                    session.id,
+                    promptRequest
+                )
+            )
+        }
+    }
+
+    override fun onWindowRequest(windowRequest: WindowRequest) {
+        if (windowRequest.type == WindowRequest.Type.CLOSE) {
+            activity?.onBackPressed()
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt]: I think `findNavController().popBackStack()` is better here to keep it consistent.

File: app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt [hunk 2]
```diff
@@ -44,7 +41,7 @@ class WebExtensionActionPopupFragment : Fragment(), EngineSession.Observer {
     ): View? {
         // Grab the [EngineSession] from the store when the view is created if it is available.
         if (engineSession == null) {
-            engineSession = coreComponents.store.state.extensions[args.webExtensionId]?.popupSession
+            createSession(coreComponents.store.state.extensions[args.webExtensionId]?.popupSession)
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt]: What about when we consume the `popupSession` from the store below?

File: app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt (hunk 1 above)
Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt]: OK, this isn't a change I made, but let me try.

File: app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt (hunk 2 above)
Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt]: What about it :) ?

Comment (psymoon) [reply app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt]: Shouldn't we make sure to dispatch `AddCustomTabAction` for that as well?

Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/addons/WebExtensionActionPopupFragment.kt]: @psymoon no, we don't need to do that, as this is handled by the base class in createView/destroyView.

**Pull request merged**